lib/processors/resourceListCreator.js

```diff
@@ -0,0 +1,697 @@
+			}
+		}
+
+		for (const baseName of baseNames) {
+			if ( this._resources.has(baseName) ) {
+				const info = this._resources.get(baseName);
+				info.i18nName = baseName;
+				info.i18nLocale = "";
+			}
+		}
+
+		return Promise.all(promises);
+	}
+
+	createOrphanFilters() {
+		log.verbose("	configured external resources filters (resources outside the namespace): %s", this.externalResources == null ? "(none)" : this.externalResources);
+
+		const filters = new Map();
+
+		if ( this._externalResources != null && !this._externalResources.isEmpty() ) {
+			/*
+			Collection<String> componentsToProcess = new HashSet<String>(Arrays.asList(this.externalResources.trim().split("\\s*,\\s*")));
+
+			for ( String component : componentsToProcess ) {
+				ResourceFilterList packageFilters = null;
+				int p = component.indexOf(':');
+				if ( p > 0 ) {
+					String packages = component.substring(p+1).trim();
+					component = component.substring(0, p).trim();
+					if ( packages != null ) {
+						packageFilters = new ResourceFilterList().addFilters(packages.trim().split("\\s*;\\s*"));
+					}
+				}
+				if ( component.equals("/") || component.isEmpty() ) {
+					component = "";
+				} else if ( !component.endsWith("/") ) {
+					component += "/";
+				}
+				log.verbose("	resulting filter list for '%s': '%s'", component, packageFilters);
+				filters.put(component, packageFilters);
+			}
+			*/
+		}
+		return filters;
+	}
+
+	groupResourcesByComponents() {
+		const orphanFilters = this.createOrphanFilters();
+
+		for (const resource of this._resources.values()) {
+			let contained = false;
+			for (const [prefix, list] of this._components.entries()) {
+				if ( resource.name.startsWith(prefix) ) {
+					list.add(resource);
+					contained = true;
+				} else if ( orphanFilters.has(prefix) ) {
+					//						if ( log.isDebug() ) {
+					//							log.verbose("	checking '%s' against orphan filter '%s' (%s)", resource.name, orphanFilters.get(prefix), prefix);
+					//						}
+					if ( orphanFilters.get(prefix).matches(resource.name) ) {
+						list.add(resource);
+						contained = true;
+					}
+				}
+			}
+
+			if ( resource.theme != null ) {
+				// assign theme resources additionally to theme packages
+				for (const [prefix, list] of this._themePackages.entries()) {
+					if ( resource.name.startsWith(prefix) ) {
+						list.add(resource);
+						contained = true;
+					}
+				}
+			}
+
+			if ( contained ) {
+				this._resources.delete(resource.name);
+			}
+		}
+	}
+
+	get resources() {
+		return new Set(this._resources.keys());
+	}
+
+	get components() {
+		return this._components;
+	}
+
+	get themePackages() {
+		return this._themePackages;
+	}
+}
+
+/**
+ * Whether the detection of orphans should result in a build failure.
+ *
+ * @since 1.29.1
+ */
+
+
+const getResourceSize = async (resource) => {
+	try {
+		const content = await resource.getString();
+		return Buffer.byteLength(content, "utf8");
+	} catch (e) {
+		// ignore error
+	}
+
```

Comment:
Isn't this the most expensive way to determine the size of the resource?

getString() has to read the resource, convert the UTF-8 byte stream to a Unicode16 char sequence; Buffer.byteLength then has to do the opposite operation just to determine the original UTF-8 byte length.

Overall performance when building `sap.ui.core` also looks very bad.

Reply:
hm already counted the buffer length but that's not correct, have to find a better way.
The statInfo would be the best, but this needs to be updated when the sources are compressed

Reply:
why / how was the buffer length not correct?

Reply:
Ok, `Buffer.prototype.length` is the allocation size, not the content length. Too bad. To me that means, `@ui5/fs.Resource` needs to know the size, if neither `statInfo` nor `Buffer` are suitable to determine it.

Reply:
Maybe for
@ui5/fs.Resource
we can use the [stats.size](https://nodejs.org/api/fs.html#fs_stats_size) property for size
but we must ensure that each content modification will trigger a size update

Reply:
Exactly, maintaining the (byte) size IMO should become a responsibility of the Resource, to get a proper abstraction.

Downside: when new content is provided as a string, this unfortunately requires either encoding the string as UTF-8 or using Buffer.byteLength.  Maybe that update of the size then should be done lazily.

Reply:
I agree the Resource should be responsible.
If a string is provided it is anyways converted to a buffer and the buffer.byteLength is just a property so there should be no performance impact.

Reply:
can use statinfo size with
https://github.com/SAP/ui5-fs/pull/253